src/components/services/LoanManagerService.ts

```diff
@@ -27,18 +54,95 @@ export const LoanManagerService = () => {
   //
 
   React.useEffect(() => {
-    const cleanup = () => {
+    const routine = () => {
       for (const loanAccount of Object.values(loanAccountMap)) {
         if (!checkLoanHasFunds(loanAccount) && loanAccount.closed) {
           dispatch(deleteLoanAccount(loanAccount))
         }
       }
     }
-    const intervalId = setInterval(cleanup, 3000)
+    const intervalId = setInterval(routine, 3000)
     return () => {
       clearInterval(intervalId)
     }
   }, [dispatch, loanAccountMap])
 
+  //
+  // Update Liquidation Threshold Push Event
+  //
+  React.useEffect(() => {
+    const routine = async () => {
+      for (const loanAccount of Object.values(loanAccountMap)) {
+        const { borrowEngine } = loanAccount
+        await waitForBorrowEngineSync(borrowEngine)
+        const { debts, collaterals } = borrowEngine
+        const filteredDebts = debts.filter(debt => !zeroString(debt.nativeAmount))
+        const filteredCollaterals = collaterals.filter(collateral => !zeroString(collateral.nativeAmount))
+        const onlyOneCollateral = filteredCollaterals.length === 1
+        // TODO: Find a less crude way to determine if a token is USD-based
+        const onlyUsdBasedDebts = filteredDebts.every(debt => (debt.tokenId != null ? USD_BASED_TOKEN_IDS.includes(debt.tokenId.toLowerCase()) : false))
+
+        if (onlyOneCollateral && onlyUsdBasedDebts) {
+          const getExchangeAmount = (tokenId: string | undefined, nativeAmount: string): string => {
+            const tokens = allTokens[borrowEngine.currencyWallet.currencyInfo.pluginId]
+            const token = tokenId != null ? tokens[tokenId] : undefined
+            const denom = token != null ? token.denominations[0] : borrowEngine.currencyWallet.currencyInfo.denominations[0]
+            return div(nativeAmount, denom.multiplier, DECIMAL_PRECISION)
+          }
+          const getCurrencyCode = (tokenId?: string): string => {
+            const tokens = allTokens[borrowEngine.currencyWallet.currencyInfo.pluginId]
+            const token = tokenId != null ? tokens[tokenId] : undefined
+            return token?.currencyCode ?? borrowEngine.currencyWallet.currencyInfo.currencyCode
+          }
+
+          const debtFiatAmounts = filteredDebts.map(debt => {
+            const debtCurrencyCode = getCurrencyCode(debt.tokenId)
+            const debtExchangeAmount = getExchangeAmount(debt.tokenId, debt.nativeAmount)
+            return convertCurrency(state, debtCurrencyCode, 'iso:USD', debtExchangeAmount) //, 2
```

Comment:
Use `convertCurrencyFromExchangeRates` and subscribe to `state.exchangeRates` instead of `state` on line 39

Reply:
Do you really even need to get the exchange rate. Might as well just assume 1-1 with USD

Reply:
I think because we have to assume an exchange rate to USD on the push-server that the correct thing is to remove the stable-coin exchange rate calculation like you said. Our assumption is already being made on the push-server and changing this assumption locally wont change that.